UndertaleModLib/Compiler/Compiler.cs

```diff
@@ -54,6 +56,22 @@ public void OnSuccessfulFinish()
                 {
                     foreach (KeyValuePair<string, string> v in GlobalVars)
                         Data?.Variables?.EnsureDefined(v.Key, UndertaleInstruction.InstanceType.Global, false, Data.Strings, Data);
+                    if (Data != null) {
+                        foreach (string name in FunctionsToObliterate) {
+                            string scriptName = "gml_Script_" + name;
+                            UndertaleScript scriptObj = Data.Scripts.ByName(scriptName);
+                            if (scriptObj != null) Data.Scripts.Remove(scriptObj);
+                            UndertaleCode codeObj = Data.Code.ByName(scriptName);
+                            if (codeObj != null) {
+                                Data.Code.Remove(codeObj);
+                                OriginalCode.ChildEntries.Remove(codeObj);
+                            }
+                            UndertaleFunction functionObj = Data.Functions.ByName(scriptName);
+                            if (functionObj != null) Data.Functions.Remove(functionObj);
+                            Data.KnownSubFunctions.Remove(name);
```

Comment:
This thing looks a bit weird. Can a scriptName be both in Data.Scripts and Data.Code ? To my knowledge assets have to have unique names.

Reply:
Yes I think, each function definition is in both Scripts and Code (Data.Scripts.ByName and Data.Code.ByName both return those entries too):
![image](https://github.com/krzys-h/UndertaleModTool/assets/68464103/917ddeda-d650-4b85-90eb-501fbea39c2b)

Reply:
huh, TIL.